sdk/core/core-client/src/urlHelpers.ts

```diff
@@ -288,11 +291,13 @@ export function appendQueryParams(
   for (const [name, value] of combinedParams) {
     if (typeof value === "string") {
       searchPieces.push(`${name}=${value}`);
-    } else {
+    } else if (Array.isArray(value)) {
       // QUIRK: If we get an array of values, include multiple key/value pairs
       for (const subValue of value) {
         searchPieces.push(`${name}=${subValue}`);
       }
+    } else {
+      searchPieces.push(`${name}=${value}`);
```

Comment:
Wait, is this designed to handle undefined? Wouldn’t it add `undefined` as the string value here?

Reply:
yes, it would add `undefined` as the literal value

Reply:
is that intentional? if so, disregard.

Reply:
some context: https://github.com/Azure/azure-sdk-for-js/pull/18621#discussion_r747748890

Reply:
@sarangan12 perhaps consider creating a test case for this behavior to document it.

Reply:
Wait, shouldn’t we skip it? would it work too or would it fail? meaning, to remove things like `C=undefined`. The link in context indicates that `C=undefined` works, but it does not indicate that removing it would fail.

Reply:
I'm a little torn, since silently removing the query params seems like a slightly larger jump than filling in a bogus value. @sarangan12 did your investigation yield if the presence of this parameter mattered at all?

Reply:
@xirzec Yes. Removing the value will cause the error if the value is not an optional value.